clang/lib/CodeGen/CGExprComplex.cpp

```diff
@@ -310,6 +310,13 @@ class ComplexExprEmitter
         CGF.getContext().getFloatTypeSemantics(ElementType);
     const llvm::fltSemantics &HigherElementTypeSemantics =
         CGF.getContext().getFloatTypeSemantics(HigherElementType);
+    // Check that LongDouble Size > Double Size.
+    // This can be interpreted as:
+    // SmallerType.LargestFiniteVal * SmallerType.LargestFiniteVal <=
+    // LargerType.LargestFiniteVal.
```

Comment:
Not entirely accurate, it's `(SmallerType.LargestFiniteVal * SmallerType.LargestFiniteVal) * 2`